src/js/tabs/pid_tuning.js

```diff
@@ -2192,3 +2298,177 @@ TABS.pid_tuning.updatePIDColors = function(clear = false) {
     setTuningElementColor($('.pid_tuning .PITCH input[name="f"]'), ADVANCED_TUNING_ACTIVE.feedforwardPitch, ADVANCED_TUNING.feedforwardPitch);
     setTuningElementColor($('.pid_tuning .YAW input[name="f"]'), ADVANCED_TUNING_ACTIVE.feedforwardYaw, ADVANCED_TUNING.feedforwardYaw);
 };
+
+TABS.pid_tuning.changeRatesType = function(rateTypeID) {
+    var self = this;
+    var dialogRatesType = $('.dialogRatesType')[0];
+    var sameRatesType = true;
+    
+    self.currentRatesType = rateTypeID;
+            
+    if (self.currentRatesType != RC_tuning.rates_type) {
+        sameRatesType = false;
+        dialogRatesType.showModal();
+
+        $('.dialogRatesType-cancelbtn').click(function() {
+            sameRatesType = true;
+            self.currentRatesType = RC_tuning.rates_type;
+            $('.rates_type select[id="ratesType"]').val(RC_tuning.rates_type);
+            changeRatesSystem();
+            dialogRatesType.close();
+        });
+
+        $('.dialogRatesType-confirmbtn').click(function() {
+            changeRatesSystem();
+            dialogRatesType.close();
+        });
+    } else {
+        changeRatesSystem();
+        return;
+    }
+    
+    function changeRatesSystem() {
+        self.changeRatesTypeLogo();
+        
+        var rc_rate_max = 2.55, rc_rate_min = 0.01, rc_rate_step = 0.01;
+        var rate_max = 1.0, rate_min = 0, rate_step = 0.01;
+        var expo_max = 1.0, expo_min = 0, expo_step = 0.01;
+        
+        var rc_rate_default = (1).toFixed(2), rate_default = (0.7).toFixed(2), expo_default = (0).toFixed(2); // default values for betaflight curve. all the default values produce the same betaflight default curve (or at least near enough)
+        
+        if (sameRatesType) { // if selected rates type is different from the saved one, set values to default instead of reading
```

Comment:
If you want to use this variable in here you should be passing it in, and not be reliant on it wrongly being defined as a global variable.